src/core/struct.c

```diff
@@ -29,12 +29,11 @@
 #endif
 
 /* Begin creation of a struct */
-JanetKV *janet_struct_begin(int32_t count) {
+JanetKV *janet_struct_begin(size_t count) {
     /* Calculate capacity as power of 2 after 2 * count. */
-    int32_t capacity = janet_tablen(2 * count);
-    if (capacity < 0) capacity = janet_tablen(count + 1);
+    size_t capacity = janet_tablen(2 * count);
 
-    size_t size = sizeof(JanetStructHead) + (size_t) capacity * sizeof(JanetKV);
+    size_t size = sizeof(JanetStructHead) + capacity * sizeof(JanetKV);
```

Comment:
The old code didn't do this as well for older platforms, but we might want to do some overflow checking with this size calculation and prevent a bad allocation.

Reply:
What should be the limit here? I thought `Janet_gcalloc` handled when malloc fails

Reply:
The issue is that the calculation of capacity * sizeof(JanetKV) could overflow in a 32 bit system if capacity is close to that the limit. The old code could over flow as well in that case, and it is hard to trigger this for pure Janet, but probably easier with the C api

Reply:
Is the last commit check ok for this case?

Reply:
We should cast to uint64_t instead of size_t and ensure we are always doing 64 bit arithmetic.